system/src/system_network_internal.h

```diff
@@ -545,6 +554,11 @@ class ManagedNetworkInterface : public NetworkInterface
                 if (ret != 0) {
                     diag->lastError(ret);
                 }
+            } else if (onRet) {
+                // on() failed, we'll rely on the system loop to call connect() again
+                // FIXME: this is a UI workaround, as otherwise the device
+                // will be in a breathing blue state, which might be confusing
+                LED_SIGNAL_START(NETWORK_CONNECTING, NORMAL);
```

Comment:
Let's not introduce UI changes so far into 2.x as the comment states, especially considering this whole PR is just piling on more hacks. This was broken previously and we were blinking green under these circumstances. I'd like to keep the same behavior in order not to cause any confusion.

Reply:
Could you add this to the comments, because it's not clear what the FIXME should correctly be sometime later.  In fact it says we should be breathing blue, not breathing white.